File: QID-1213-MVAincomeLi/MVAincomeLi.r

# clear all variables
rm(list = ls(all = TRUE))
graphics.off()

# install and load packages
libraries = c("hexbin")
lapply(libraries, function(x) if (!(x %in% installed.packages())) {
    install.packages(x)
})
lapply(libraries, library, quietly = TRUE, character.only = TRUE)

# load data
allbus  = read.csv2("allbus.csv")
allbus7 = allbus[, c(3, 8)]

# exclude unvalid observations
allbus7       = allbus7[(allbus7$NETTOEIN < 99997) & (allbus7$WOHNFLAE < 998), ]
living_space  = allbus7[, 2]
netincome     = allbus7[, 1]

# hexagon plot
hexbinplot(living_space ~ netincome, main="Hexagon plot", xlab = "Income", ylab = "Flat Size", style = "colorscale", 
    border = TRUE, aspect = 1, trans = sqrt, inv = function(netincome) netincome^2)

File: QID-1213-MVAincomeLi/MVAincomeLi.py
#works on pandas 1.5.2 and matplotlib 3.6.2
import pandas as pd
import matplotlib.pyplot as plt

# Load data
allbus = pd.read_csv("allbus.csv", sep=";", decimal=",")
allbus1 = allbus.iloc[:, [2,7]]

# Exclude invalid observations
allbus1 = allbus1[(allbus1['NETTOEIN'] < 99997) & (allbus1['WOHNFLAE'] < 998)]

# Get ages and net income
netincome = allbus1.iloc[:, 0]
living_space = allbus1.iloc[:, 1]

#create plot
fig, ax = plt.subplots(figsize=(12,10))
hb = ax.hexbin(netincome, living_space, gridsize=30, cmap='Blues',
                edgecolors = 'black', mincnt = 1)
ax.set_title("Hexagon plot", fontweight = 'bold', fontsize = 30, pad = 15)
ax.set_xlabel("Income", fontsize = 25)
ax.set_ylabel("Flat Size", fontsize = 25)
ax.tick_params(axis='both', labelsize=25)
cbar = fig.colorbar(hb, drawedges = True) 
cbar.ax.tick_params(labelsize=22)

plt.show()